src/main/java/com/google/sps/servlets/InitializeServlet.java

```diff
@@ -20,7 +23,24 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         
-    //TODO - implement the rest of this function (@Kaleab)
+    String correlator = request.getParameter("correlator");
+
+    //TODO: find a more graceful way to handle this
+    if (correlator == null){
+      throw new IOException();
+    }
+
+    Entity impressions = new Entity("Impressions");
+    impressions.setProperty("correlator", correlator);
```

Comment:
all these should be consts

Reply:
When you say these, are you referring to the properties themselves?